protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -145,11 +145,14 @@ func getEnumDefault(reg *descriptor.Registry, enum *descriptor.Enum) string {
 }
 
 // messageToQueryParameters converts a message to a list of OpenAPI query parameters.
-func messageToQueryParameters(message *descriptor.Message, reg *descriptor.Registry, pathParams []descriptor.Parameter, body *descriptor.Body) (params []openapiParameterObject, err error) {
+func messageToQueryParameters(message *descriptor.Message, reg *descriptor.Registry, pathParams []descriptor.Parameter, body *descriptor.Body, httpMethod string) (params []openapiParameterObject, err error) {
 	for _, field := range message.Fields {
 		if !isVisible(getFieldVisibilityOption(field), reg) {
 			continue
 		}
+		if field.GetTypeName() == ".google.protobuf.FieldMask" && reg.GetAllowPatchFeature() && httpMethod == "PATCH" && len(body.FieldPath) != 0 {
```

Comment:
I think we also want to check that the field name is `update_mask` exactly. Also, we should check the flag first, since it invalidates everything else
```suggestion
		if reg.GetAllowPatchFeature() && field.GetTypeName() == ".google.protobuf.FieldMask" && httpMethod == "PATCH" && len(body.FieldPath) != 0 {
```

Reply:
ok